src/js/components/MenuComponent.jsx

```diff
@@ -0,0 +1,49 @@
+import React from "react/addons";
+
+import Util from "../helpers/Util";
+
+import MenuItemComponent from "../components/MenuItemComponent";
+
+var MenuComponent = React.createClass({
+  "displayName": "MenuComponent",
+
+  propTypes: {
+    children: Util.getComponentTypeValidator(MenuItemComponent),
+    className: React.PropTypes.string,
+    name: React.PropTypes.string,
+    onChange: React.PropTypes.func,
+    selected: React.PropTypes.string
+  },
+
+  getDefaultProps: function () {
+    return {
+      name: "menu-" + Util.getUniqueId(),
+      onChange: Util.noop
+    };
+  },
+
+  renderChildren: function () {
+    var {children, name, selected} = this.props;
+
+    return React.Children.map(children,  (child) =>
+      React.addons.cloneWithProps(child, {
```

Comment:
Not sure if we care at all, but this is deprecated - https://facebook.github.io/react/docs/clone-with-props.html